BlueprintUICommonControls/Tests/Sources/Internal/KeyboardObserverTests.swift

```diff
@@ -0,0 +1,182 @@
+            XCTAssertEqual(info.animationDuration, 2.5)
+            XCTAssertEqual(info.animationCurve, 123)
+        }
+        
+        self.testcase("Failed Inits") {
+            
+            self.testcase("No userInfo") {
+                self.assertThrowsError(test: {
+                    try _ = KeyboardObserver.NotificationInfo(
+                        with: Notification(name: UIResponder.keyboardDidShowNotification, object: nil, userInfo: nil)
+                    )
+                }, verify: { error in
+                    XCTAssertEqual(error as? KeyboardObserver.NotificationInfo.ParseError, .missingUserInfo)
+                })
+            }
+            
+            self.testcase("No end frame") {
+                
+                var userInfo = defaultUserInfo
+                userInfo.removeValue(forKey: UIResponder.keyboardFrameEndUserInfoKey)
+                
+                self.assertThrowsError(test: {
+                    try _ = KeyboardObserver.NotificationInfo(
+                        with: Notification(name: UIResponder.keyboardDidShowNotification, object: nil, userInfo: userInfo)
+                    )
+                }, verify: { error in
+                    XCTAssertEqual(error as? KeyboardObserver.NotificationInfo.ParseError, .missingEndingFrame)
+                })
+            }
+            
+            self.testcase("No animation duration") {
+                
+                var userInfo = defaultUserInfo
+                userInfo.removeValue(forKey: UIResponder.keyboardAnimationDurationUserInfoKey)
+                
+                self.assertThrowsError(test: {
+                    try _ = KeyboardObserver.NotificationInfo(
+                        with: Notification(name: UIResponder.keyboardDidShowNotification, object: nil, userInfo: userInfo)
+                    )
+                }, verify: { error in
+                    XCTAssertEqual(error as? KeyboardObserver.NotificationInfo.ParseError, .missingAnimationDuration)
+                })
+            }
+            
+            self.testcase("No animation curve") {
+                
+                var userInfo = defaultUserInfo
+                userInfo.removeValue(forKey: UIResponder.keyboardAnimationCurveUserInfoKey)
+                
+                self.assertThrowsError(test: {
+                    try _ = KeyboardObserver.NotificationInfo(
+                        with: Notification(name: UIResponder.keyboardDidShowNotification, object: nil, userInfo: userInfo)
+                    )
+                }, verify: { error in
+                    XCTAssertEqual(error as? KeyboardObserver.NotificationInfo.ParseError, .missingAnimationCurve)
+                })
+            }
+            
+        }
+    }
+}
+
+
+extension XCTestCase
+{
+    func testcase(_ name : String = "", _ block : () -> ())
+    {
+        block()
+    }
+    
+    func assertThrowsError(test : () throws -> (), verify : (Error) -> ())
```

Comment:
XCTAssertThrowsError?

Reply:
Good point. I've just been mindlessly copy/pasting this around since before that existed